tests/contrib/operators/test_r_operator.py

```diff
@@ -0,0 +1,171 @@
+        """Fail if invalid rscript_bin supplied"""
+
+        try:
+            expected_error = FileNotFoundError
+        except NameError:
+            # py2
+            expected_error = OSError
+
+        task = ROperator(
+            task_id='test_r_bad_rscript',
+            r_command='print(Sys.Date())',
+            rscript_bin='somebadrscript',
+            dag=self.dag
+        )
+
+        self.assertIsNotNone(task)
+
+        ti = TaskInstance(task=task, execution_date=timezone.utcnow())
+
+        with self.assertRaises(expected_error):
+            ti.run()
+
+    def test_xcom_output(self):
+        """Test whether Xcom output is produced using last line"""
+
+        self.task_xcom.xcom_push = True
+
+        ti = TaskInstance(
+            task=self.task_xcom,
+            execution_date=timezone.utcnow()
+        )
+
+        ti.run()
+        self.assertIsNotNone(ti.duration)
+
+        self.assertEqual(
+            ti.xcom_pull(task_ids=self.task_xcom.task_id, key='return_value'),
+            self.xcom_test_str
+        )
+
+    def test_xcom_none(self):
+        """Test whether no Xcom output is produced when push=False"""
+
+        self.task_xcom.xcom_push = False
+
+        ti = TaskInstance(
+            task=self.task_xcom,
+            execution_date=timezone.utcnow(),
+        )
+
+        ti.run()
+        self.assertIsNotNone(ti.duration)
+        self.assertIsNone(ti.xcom_pull(task_ids=self.task_xcom.task_id))
+
+    def test_env_vars(self):
+        """Test whether environment is passed properly"""
+
+        test_var = 'TEST_VALUE_X'
+        test_str = 'Hello Airflow'
+
+        task = ROperator(
+            task_id='test_env_vars',
+            r_command='cat(Sys.getenv("{}"))'.format(test_var),
+            env={test_var: test_str, "PATH": os.environ['PATH']},
+            xcom_push=True,
+            dag=self.dag
+        )
+
+        ti = TaskInstance(task=task, execution_date=timezone.utcnow())
+
+        ti.run()
+        self.assertIsNotNone(ti.duration)
+
+        self.assertEqual(
+            ti.xcom_pull(task_ids=task.task_id, key='return_value'),
+            test_str
+        )
+
+    def test_command_template(self):
+        """Test whether templating works properly with r_command"""
+
+        task = ROperator(
+            task_id='test_cmd_template',
+            r_command='cat("{{ ds }}")',
+            dag=self.dag
+        )
+
+        ti = TaskInstance(task=task, execution_date=DEFAULT_DATE)
+        ti.render_templates()
+
+        self.assertEqual(
+            ti.task.r_command,
+            'cat("{}")'.format(DEFAULT_DATE.date().isoformat())
+        )
+
+    def test_env_templates(self):
+        """Test whether templating works properly with env vars"""
+
+        test_var = 'TEST_CURR_DATE'
+
+        task = ROperator(
+            task_id='test_env_template',
+            r_command='cat(Sys.getenv("{}"))'.format(test_var),
+            env={test_var: "{{ ds }}"},
```

Comment:
The operator doesn't accept these (yet) right? Looks useful though.

Reply:
This is a remnant of a previous version. Originally, the operator ran the R code through the shell via `Rscript`.